### PR TITLE
Makefile - fix small MKL linking error

### PR DESCRIPTION
**Pull request by jeremylt**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -298,7 +298,7 @@ ifneq ($(wildcard $(XSMM_DIR)/lib/libxsmm.*),)
   ifeq (,$(MKL)$(MKLROOT))
     BLAS_LIB = -lblas
   else
-    BLAS_LIB = $(if $(MKLROOT),-L$(MKLROOT)/lib/intel64 -Wl,-rpath,$(MKLROOT)/lib/intel64) -Wl,--no-as-needed -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl
+    BLAS_LIB = $(if $(MKLROOT),-L$(MKLROOT)/lib/intel64 -Wl,-rpath,$(MKLROOT)/lib/intel64 -Wl),--no-as-needed -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl
```
Comment (jedbrown) [inline Makefile]: I think the issue is that the commas are controls for the `$(if`. Quotes or a variable containing `-L$(MKLROOT)/lib/intel64 -Wl,-rpath,$(MKLROOT)/lib/intel64` should fix it.

Comment (jedbrown) [reply Makefile]: Quotes won't do:
```
$ make print-BLAS_LIB MKLROOT=/foo
[ variable name]: BLAS_LIB
[        origin]: file
[         value]: $(if $(MKLROOT),"-L$(MKLROOT)/lib/intel64 -Wl,-rpath,$(MKLROOT)/lib/intel64 -Wl"),--no-as-needed -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl
[expanded value]: "-L/foo/lib/intel64 -Wl,--no-as-needed -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl
```

This is missing the rpath part. Just put it in a separate variable.

Comment (jeremylt) [reply Makefile]: Thanks, I hadn't tested it yet. I'll fix it up in the morning.

**Comment by jeremylt** [comment]
That's what I thought, but GCC did not like `--no-as-needed` without the `-L...`

**Comment by jedbrown** [comment]
What exactly did you try?

**Pull request merged**